framework/src/functions/LinearCombinationFunction.C

```diff
@@ -61,3 +61,21 @@ LinearCombinationFunction::value(Real t, const Point & p)
     val += _w[i] * _f[i]->value(t, p);
   return val;
 }
+
+RealGradient
+LinearCombinationFunction::gradient(Real t, const Point & p)
+{
+  RealGradient g;
+  for (unsigned i = 0; i < _f.size(); ++i)
+    g += _w[i] * _f[i]->gradient(t, p);
+  return g;
+}
+
+RealVectorValue
+LinearCombinationFunction::vectorValue(Real t, const Point & p)
```

Comment:
I think I just didn't like the fact that if I had a function that returned a vector that I needed do use a method called "gradient", when I didn't want a gradient just a vector function.

That is, it was for code read ability, calling gradient when you don't want the gradient seemed wrong.